examples/tests/invalid/class.w

```diff
@@ -78,3 +78,83 @@ class C8 extends S1 {
 class C11 extends C11 {
                 //^^^ Class cannot extend itself
 }
+
+class Student {
+  name: str;
+  major: str;
+  init(name: str, major: str) {
+    super();
+//  ^^^^^^^^ Calls to super constructor can only be made from derived classes
+    this.name = name;
+    this.major = major;
+  }
+}
+
+class PaidStudent extends Student {
+  hrlyWage: num;
+  
+  init(name: str, major: str, hrlyWage: num) {
+    this.hrlyWage = hrlyWage;
+    super(name, major);
+//  ^^^^^^^^^^^^^^^^^^^ Expected call to super to be first statement in constructor
+  }
+
+  something() {
+    super("cool", "blue");
+//  ^^^^^^^^^^^^^^^^^^^^^^ Calls to super constructor can only be done from within class constructor
+  }
+}
+
+   super();
+// ^^^^^^^^ Calls to super constructor can only be done from within a class constructor
+let s = new PaidStudent("x", "y", 3);
+
+// Typechecking super calls
+class One {
+  someNum: num;
+  init(someNum: num) {
+    this.someNum = someNum;
+  }
+}
+
+class Two extends One {
+  someStr: str;
+  init(someNum: num, someStr: str) {
+    super(someStr);
+//        ^^^^^^^ Expected type to be "num", but got "str" instead
+    this.someStr = someStr;
+  }
+}
+
+class Three extends One {
+  someStr: str;
+  init(someNum: num, someStr: str) {
+    super();
+//  ^^^^^^^^ Expected 1 positional argument(s) but got 0
+    this.someStr = someStr;
+  }
+}
+
+class Four extends One {
+  someStr: str;
+  init(someNum: num, someStr: str) {
+    super(someNum, someStr);
+//  ^^^^^^^^^^^^^^^^^^^^^^^^ Expected 1 arguments but got 2
```

Comment:
I suspect this will also work for keyword arguments, correct?

Reply:
I think it should since the type checking is the same but I forgot to make an example. Ill add now

Reply:
Looks like theres an issue with keyword args, looking into it.